### PR TITLE
feat: add letter case action

### PR DESCRIPTION
**Pull request by bjarneo**
Check README.md for the full documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
-# kebab-case-action
-A Public GitHub Action for converting things to kebab-case
+# Letter case action
+
+<p align="center">
+  <a href="https://github.com/dignio/letter-case-action/actions"><img alt="letter-case-action status" src="https://github.com/dignio/letter-case-action/workflows/testing/badge.svg"></a>
```
Comment (bjarneo) [inline README.md]: hm, must find correct URL here

File: README.md [hunk 2]
```diff
@@ -1,2 +1,72 @@
...
+
+# For tests
+npm test
+
+# Testing the compiled script (https://github.com/vercel/ncc)
+npm run build
+INPUT_STRING=feature/branch-1337 node dist/index.js
+```
+
+## Package for distribution
```
Comment (bjarneo) [inline README.md]: Remove this section

File: index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+function createCases(inputString) {
+    return {
+        kebab: kebabCase(inputString),
+        camel: camelCase(inputString),
+        snake: snakeCase(inputString),
+    };
+}
+
+(async function main() {
```
Comment (bjarneo) [inline index.js]: remove async and IIFE function

File: .github/workflows/check-dist.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+  workflow_dispatch:
```
Comment (bjarneo) [inline .github/workflows/check-dist.yml]: remove this one

File: .github/workflows/check-dist.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+name: Check if the dist has a diff
+
+on:
+  push:
```
Comment (bjarneo) [inline .github/workflows/check-dist.yml]: remove push + branch

File: .github/workflows/check-dist.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Set Node.js 16.x
+        uses: actions/setup-node@v2.5.1
+        with:
+          node-version: 16.x
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Rebuild the dist/ directory
+        run: npm run prepare
```
Comment (bjarneo) [inline .github/workflows/check-dist.yml]: change to npm run build as prepare is not used anymore

File: .github/workflows/check-dist.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Compare the expected and actual dist/ directories
+        run: |
+          if [ "$(git diff --ignore-space-at-eol dist/ | wc -l)" -gt "0" ]; then
+            echo "Detected uncommitted changes after build.  See status below:"
+            git diff
+            exit 1
+          fi
+        id: diff
+
+      # If index.js was different than expected, upload the expected version as an artifact
```
Comment (bjarneo) [inline .github/workflows/check-dist.yml]: remove the upload of the dist

File: .github/workflows/check-dist.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Rebuild the dist/ directory
+        run: npm run prepare
+
+      - name: Compare the expected and actual dist/ directories
+        run: |
+          if [ "$(git diff --ignore-space-at-eol dist/ | wc -l)" -gt "0" ]; then
+            echo "Detected uncommitted changes after build.  See status below:"
```
Comment (bjarneo) [inline .github/workflows/check-dist.yml:31]: Add a comment on how to solve this

Comment (bjarneo) [reply .github/workflows/check-dist.yml:31]: maybe add an action to add this to the PR

Comment (bjarneo) [reply .github/workflows/check-dist.yml:31]: https://github.com/marketplace/actions/add-pr-comment

**Comment by bjarneo** [comment]
Good to go!

**Review by lemonsaurus — APPROVED** [review]
Number #1 pull request deserves #1 prize.🥇

**Pull request merged**